Thoughtless-Eels/Bangazon-WebApp
# Refactor product model & new migration

## Link to Ticket
no ticket

## Description of Proposed Changes

- Refactored product model
- cleared out migration folder and performed a new fresh migration

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout gl-refactorProductModel

```

- Delete all tables from Database through ServerExplorer
- Database file should be called "aspnet-BangazonWebApp.mdf and should be located in your Home folder
- Run `Update-Database` in the Project Manager Console and check for errors.


## Impacted Areas in Application

List general components of the application that this PR will affect:

* database
* Product.cs

## Mentions @username

Tag users that need to review this code

## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each.
